### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by Bludidi**
## In this pull request I have implemented the following: 

- Create a `app.rb` file that has the following methods

  - [ ] List all books.
  - [ ] List all people.
  - [ ]  Create a person (teacher or student, not a plain Person).
  - [ ]  Create a book.
  - [ ]  Create a rental.
  - [ ]  List all rentals for a given person id.

- Create `main.rb` and define an entry point called main to:

  - [ ] Present the user with a list of options to perform.
  - [ ] Let users choose an option.
  - [ ] If needed, ask for parameters for the option.
  - [ ] Have a way to quit the app.

### CODE REVIEW

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @Bludidi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## ❌ Invalid Code Review Request

You have submitted a project, that does not have at least two future features that you will add to your project in the future. Kindly add this to the future features section of your readme. Also, add key features of your app so as to make your project professional.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: App/app.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_num = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+    @rentals_list.push(Rental.new(date, @books_list[book_num], @people_list[person_num]))
+    puts 'Rental created successfully'
+    @parent.show_menu
+  end
```
Comment (Whoistolu) [inline App/app.rb]: - [ ] [REQUIRED] Great job @Bludidi 👍🏽 However, whenever I try to create a rental I encounter an error. Let's take a look at the `create_rental` function to fix that.
Screenshot 2023-01-14 at 11 50 44

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Bludidi 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: App/app.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    end
+
+    person_num = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+    @rentals_list.push(Rental.new(date, @books_list[book_num], @people_list[person_num]))
+    puts 'Rental created successfully'
+    @parent.show_menu
+  end
```
Comment (rotimiazeez) [inline App/app.rb]: - Please note that I still get an error when I try to create a rental. I suggest you read through the error message to fix it. 😉
Screenshot from 2023-01-14 21-51-14

File: App/app.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    if @people_list.empty?
+      puts 'No record found! Add a person...'
+    else
+      puts 'All people in the library'
+      @people_list.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+    @parent.show_menu
+  end
```
Comment (rotimiazeez) [inline App/app.rb]: - Please note that I encounter some errors when I try to list all people. I'd suggest you check the error message for more insight into what went wrong 👍🏽.
Screenshot from 2023-01-14 22-01-17

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Bludidi 😉, 
## To Highlight 👇🏽 
- Good App UI 🖼️ 
- No Linter Errors ✔️ 
- Good PR descriptions 👍🏽
- Good Commit messages 👍🏽 
Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by prolajumokeoni — APPROVED** [review]
Hi @Bludidi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good Readme 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**